service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java

```diff
@@ -327,207 +277,212 @@ public DeviceTwinDevice replaceTwin(DeviceTwinDevice device) throws IotHubExcept
-        return new QueryCollection(sqlQuery, pageSize, QueryType.TWIN, this.iotHubConnectionString, this.iotHubConnectionString.getUrlTwinQuery(), HttpMethod.POST, options.getHttpConnectTimeout(), options.getHttpReadTimeout(), proxy);
+        Proxy proxy = options.getProxyOptions() == null
+                ? null
+                : options.getProxyOptions().getProxy();
+        return new QueryCollection(
+                sqlQuery,
+                pageSize,
+                QueryType.TWIN,
+                this.iotHubConnectionString,
+                this.iotHubConnectionString.getUrlTwinQuery(),
+                HttpMethod.POST,
+                options.getHttpConnectTimeout(),
+                options.getHttpReadTimeout(),
+                proxy);
     }
 
     /**
      * Returns the availability of next twin element upon query. If non was found,
      * Query is sent over again and response is updated accordingly until no response
      * for the query was found.
      *
-     * @param deviceTwinQuery Query object returned upon creation of query.
-     * @return True if next is available and false other wise.
-     * @throws IotHubException If IotHub could not respond back to the query successfully.
+     * @param deviceTwinQuery The Query object returned upon creation of query.
+     * @return {@code True} if next is available and {@code false} otherwise.
+     * @throws IotHubException If Iot hub could not respond back to the query successfully.
      * @throws IOException If input parameter is incorrect.
      */
     public synchronized boolean hasNextDeviceTwin(Query deviceTwinQuery) throws IotHubException, IOException
     {
         if (deviceTwinQuery == null)
         {
-            //Codes_SRS_DEVICETWIN_25_053: [ The method shall throw IllegalArgumentException if query is null ]
-            throw new IllegalArgumentException("Query cannot be null");
+            throw new IllegalArgumentException("Query cannot be null.");
         }
 
-        //Codes_SRS_DEVICETWIN_25_055: [ If a queryResponse is available, this method shall return true as is to the user, and false otherwise.. ]
         return deviceTwinQuery.hasNext();
     }
 
     /**
      * Returns the next device twin document.
      *
-     * @param deviceTwinQuery Object corresponding to the query in request.
+     * @param deviceTwinQuery The object corresponding to the query in request.
      * @return Returns the next device twin document.
-     * @throws IOException If input parameter is incorrect.
-     * @throws IotHubException If a non successful response from IotHub is received.
+     * @throws IOException If the input parameter is incorrect.
+     * @throws IotHubException If an unsuccessful response from Iot Hub is received.
      * @throws NoSuchElementException If no additional element was found.
      */
     public synchronized DeviceTwinDevice getNextDeviceTwin(Query deviceTwinQuery) throws IOException, IotHubException, NoSuchElementException
```

Comment:
What is an acceptable boundary to split this into multple lines, given it exceeds the 120 character max?

Reply:
I'd put the "throws" statement on an indented new line

Reply:
```java
public void myReallyLongMethodNameThatIsAbout120Characters(...)
  throws IOException, TransportException {

}
```

Reply:
I've always wondered when I should break the line. I usually do it at 150 chars.

Reply:
IntelliJ has a vertical line at 120 characters. I assumed (poorly?) that was our standard or at least a Java standard. What say you, @timtay-microsoft?